tools/federation/src/test/java/org/eclipse/rdf4j/federated/FedXBaseTest.java

```diff
@@ -83,14 +81,18 @@ public static void initLogging() throws Exception {
 	 * @param checkOrder
 	 * @throws Exception
 	 */
-	protected void execute(RepositoryConnection conn, String queryFile, String expectedResultFile, boolean checkOrder)
+	protected void execute(String queryFile, String expectedResultFile, boolean checkOrder)
 			throws Exception {
 
 		String queryString = readQueryString(queryFile);
 
 		Query query = queryManager().prepareQuery(queryString);
 
 		if (query instanceof TupleQuery) {
+			// Some query results will automatically close themselves when they are exhausted. To properly test that
+			// query results are closed correctly we need to evaluate the query without retrieving any elements.
+			((TupleQuery) query).evaluate().close();
+
```

Comment:
This is one of the two main changes to the tests that make the unclosed iterations visible.